extensions/tmtv/src/utils/calculateSUVPeak.ts

```diff
@@ -0,0 +1,152 @@
+    };
+    cachedStats?: {
+      projectionPoints?: Types.Point3[][];
+    };
+  };
+};
+
+/**
+ * This method calculates the SUV peak on a segmented ROI from a reference PET
+ * volume. If a rectangle annotation is provided, the peak is calculated within that
+ * rectangle. Otherwise, the calculation is performed on the entire volume which
+ * will be slower but same result.
+ * @param viewport Viewport to use for the calculation
+ * @param labelmap Labelmap from which the mask is taken
+ * @param referenceVolume PET volume to use for SUV calculation
+ * @param toolData [Optional] list of toolData to use for SUV calculation
+ * @param segmentIndex The index of the segment to use for masking
+ * @returns
+ */
+function calculateSuvPeak(
+  labelmap: Types.IImageVolume,
+  referenceVolume: Types.IImageVolume,
+  annotations?: AnnotationsForThresholding[],
+  segmentIndex = 1
+): {
+  max: number;
+  maxIJK: Types.Point3;
+  maxLPS: Types.Point3;
+  mean: number;
+} {
+  if (referenceVolume.metadata.Modality !== 'PT') {
+    return;
+  }
+
+  if (labelmap.scalarData.length !== referenceVolume.scalarData.length) {
+    throw new Error(
+      'labelmap and referenceVolume must have the same number of pixels'
+    );
+  }
+
+  const {
+    scalarData: labelmapData,
+    dimensions,
+    imageData: labelmapImageData,
+  } = labelmap;
+
+  const {
+    scalarData: referenceVolumeData,
+    imageData: referenceVolumeImageData,
+  } = referenceVolume;
+
+  let boundsIJK;
+  // Todo: using the first annotation for now
+  if (annotations && annotations[0].data?.cachedStats) {
+    const { projectionPoints } = annotations[0].data.cachedStats;
+    const pointsToUse = [].concat(...projectionPoints); // cannot use flat() because of typescript compiler right now
+
+    const rectangleCornersIJK = pointsToUse.map(world => {
+      const ijk = vec3.fromValues(0, 0, 0);
+      referenceVolumeImageData.worldToIndex(world, ijk);
+      return ijk as Types.Point3;
+    });
+
+    boundsIJK = utilities.boundingBox.getBoundingBoxAroundShape(
+      rectangleCornersIJK,
+      dimensions
+    );
+  }
+
+  let max = 0;
+  let maxIJK = [0, 0, 0];
+  let maxLPS = [0, 0, 0];
+
+  const callback = ({ pointIJK, pointLPS }) => {
+    const offset = referenceVolumeImageData.computeOffsetIndex(pointIJK);
+    const value = labelmapData[offset];
+
+    if (value !== segmentIndex) {
+      return;
+    }
+
+    const referenceValue = referenceVolumeData[offset];
+
+    if (referenceValue > max) {
+      max = referenceValue;
+      maxIJK = pointIJK;
+      maxLPS = pointLPS;
+    }
+  };
+
+  utilities.pointInShapeCallback(
+    labelmapImageData,
+    () => true,
+    callback,
+    boundsIJK
+  );
+
+  const direction = labelmapImageData
+    .getDirection()
+    .slice(0, 3) as Types.Point3;
+  /**
+   * 2. Find the bottom and top of the great circle for the second sphere (1cc sphere)
+   * diameter of 12mm = 1.2cm ~ = 1cc sphere
+   */
+  const diameter = 12;
```

Comment:
Can you make this a constant outside this file and actually calculate it correctly once at run time? i.e. 2 * ³⎷(3 * Volume / (4 π))